spec/syntax.md

```diff
@@ -834,14 +840,14 @@ _option values_, and _keys_)
 _Messages_ that contain right-to-left (aka RTL) characters SHOULD use one of the 
 following mechanisms to make messages display intelligibly in plain-text editors:
 
-1. Use paired isolating bidi controls `U+2066 LEFT-TO-RIGHT ISOLATE`
-   and `U+2069 POP DIRECTIONAL ISOLATE` as permitted by the ABNF around
+1. Use paired isolating bidi controls `U+2066 LEFT-TO-RIGHT ISOLATE` ("LRI")
+   and `U+2069 POP DIRECTIONAL ISOLATE` ("PDI") as permitted by the ABNF around
    parts of any _message_ containing RTL characters:
    - _inside_ of _placeholder_ markers `{` and `}` 
    - _outside_ _quoted-pattern_ markers `{{` and `}}`
-   - _identifiers_
-   - _literals_ (This is especially important for individual _keys_ in a _variant_)
-   - _option_ values
+   - _outside_ of _literals_, paying particular attention to _keys_ in a _variant_
+   - _outside_ of _variable_, _function_, _markup_, or _attribute_ _names_/_identifiers_,
+     including the identifying sigil (e.g. `<LRI>$var</PDI>` or `<LRI>:ns:name</PDI>`)
```

Comment:
I still don't understand a recommendation to isolate right-to-left content as left-to-right. It makes sense for stuff with prefixes (_variable_, _function_, _markup_, and _attribute_) if the surrounding context isn't LTR, but isn't it going to break the rendering of RTL _literals_ and _names_ that start with neutral characters?

Reply:
> ... but isn't it going to break the rendering of RTL literals and names that start with neutral characters?

Yes, if it were used by itself. But (2) covers that. Using LRI/PDI gets the complete "token" isolated, with the sigil on the left side and the tokens themselves ordered LTR. Using strong marks _inside_ the token (before/after `name` or unquoted literals) then fixes problems with neutrals and weaks.

Reply:
I get that for the productions which include a leading sigil. But I do not understand the recommendation to add LRI..PDI isolation "_outside_ of _literals_" or "_outside_ of _variable_, _function_, _markup_, or _attribute_ _names_/_identifiers_". These productions do not include a leading sigil which is expected to stay on the left. They only include _literal_, _name_, or _identifier_.

I think we should recommend here something like this instead:
```suggestion
   - _outside_ of _variable_, _function_, _markup_, or _attribute_,
     including the identifying sigil (e.g. `<LRI>$var</PDI>` or `<LRI>:ns:name</PDI>`)
```

Reply:
> "outside of variable, function, markup, or attribute names/identifiers"

This is `$foo`, `:foo`, `#foo`/`/foo`, and `@foo` values. They need an LRI to keep bidi from reordering them. Your suggestion is an improvement, though, as "identifier" confuses things.